core/src/test/scala/unit/kafka/utils/JsonTest.scala

```diff
@@ -45,22 +49,64 @@ class JsonTest {
+    assertEquals("1", Json.encodeAsString(1.toByte))
+    assertEquals("1", Json.encodeAsString(1.toShort))
+    assertEquals("1.0", Json.encodeAsString(1.0))
+    assertEquals("\"str\"", Json.encodeAsString("str"))
+    assertEquals("true", Json.encodeAsString(true))
+    assertEquals("false", Json.encodeAsString(false))
+    assertEquals("[]", Json.encodeAsString(Seq().asJava))
+    assertEquals("[null]", Json.encodeAsString(Seq(null).asJava))
+    assertEquals("[1,2,3]", Json.encodeAsString(Seq(1,2,3).asJava))
+    assertEquals("[1,\"2\",[3],null]", Json.encodeAsString(Seq(1,"2",Seq(3).asJava,null).asJava))
+    assertEquals("{}", Json.encodeAsString(Map().asJava))
+    assertEquals("{\"a\":1,\"b\":2,\"c\":null}", Json.encodeAsString(Map("a" -> 1, "b" -> 2, "c" -> null).asJava))
+    assertEquals("{\"a\":[1,2],\"c\":[3,4]}", Json.encodeAsString(Map("a" -> Seq(1,2).asJava, "c" -> Seq(3,4).asJava).asJava))
+    assertEquals("{\"a\":[1,2],\"b\":[3,4],\"c\":null}", Json.encodeAsString(Map("a" -> Seq(1,2).asJava, "b" -> Seq(3,4).asJava, "c" -> null).asJava))
+  }
+
+  @Test
+  def testEncodeAsBytes() {
+    assertArrayEquals("null".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(null))
+    assertArrayEquals("1".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(1))
+    assertArrayEquals("1".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(1L))
+    assertArrayEquals("1".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(1.toByte))
+    assertArrayEquals("1".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(1.toShort))
+    assertArrayEquals("1.0".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(1.0))
+    assertArrayEquals("\"str\"".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes("str"))
+    assertArrayEquals("true".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(true))
+    assertArrayEquals("false".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(false))
+    assertArrayEquals("[]".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Seq().asJava))
+    assertArrayEquals("[null]".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Seq(null).asJava))
+    assertArrayEquals("[1,2,3]".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Seq(1,2,3).asJava))
+    assertArrayEquals("[1,\"2\",[3],null]".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Seq(1,"2",Seq(3).asJava,null).asJava))
+    assertArrayEquals("{}".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Map().asJava))
+    assertArrayEquals("{\"a\":1,\"b\":2,\"c\":null}".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Map("a" -> 1, "b" -> 2, "c" -> null).asJava))
+    assertArrayEquals("{\"a\":[1,2],\"c\":[3,4]}".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Map("a" -> Seq(1,2).asJava, "c" -> Seq(3,4).asJava).asJava))
+    assertArrayEquals("{\"a\":[1,2],\"b\":[3,4],\"c\":null}".getBytes(StandardCharsets.UTF_8), Json.encodeAsBytes(Map("a" -> Seq(1,2).asJava, "b" -> Seq(3,4).asJava, "c" -> null).asJava))
```

Comment:
It's better to call `getString` on the bytes (i.e. the second argument) as that will make it easier to debug.

Reply:
Agreed, fixed.